sgkit/distance/api.py

```diff
@@ -0,0 +1,62 @@
+import numpy as np
+
+from sgkit.distance import metrics
+from sgkit.distance.core import pairwise
+from sgkit.typing import ArrayLike
+
+
+def pdist(
+    x: ArrayLike,
+    metric: str = "euclidean",
+) -> np.ndarray:
+    """Calculates the pairwise distance between all pairs of vectors in the
+    given two dimensional array x. The API is similar to:
+    ``scipy.spatial.distance.pdist``.
+
+
+    Parameters
+    ----------
+    x
+        [array-like, shape: (M, N)]
+        An array like two dimensional matrix
```

Comment:
It's worth mentioning which dimension the vectors are in that will be compared (i.e. the rows).